stingray/crossspectrum.py

```diff
@@ -1566,6 +1566,18 @@ def _initialize_from_any_input(
         ``crossspectrum_from_XXXX`` function, and initialize ``self`` with
         the correct attributes.
         """
+        if segment_size is None and isinstance(data1, StingrayObject):
+            common_gti = cross_two_gtis(data1.gti, data2.gti)
+            data1.gti = common_gti
+            data2.gti = common_gti
+            data1 = data1.apply_gtis(inplace=False)
```

Comment:
Is there a reason why you don't use `inplace=True` here, since you're storing the output in exactly the same variable?

Reply:
Because I'm creating a copy and giving it the same name, which is different from modifying the input variable. I want to avoid side effects